File: src/test/java/io/binac/leetcode/FirstMissingPositiveTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FirstMissingPositiveTest {
    private final FirstMissingPositive.Solution1 solution1 = new FirstMissingPositive.Solution1();

    @Test
    void test1() {
        int nums1[] = {1, 2, 0};
        assertEquals(3, solution1.firstMissingPositive(nums1));
        int nums2[] = {3, 4, -1, 1};
        assertEquals(2, solution1.firstMissingPositive(nums2));
        int nums3[] = {7, 8, 9, 11, 12};
        assertEquals(1, solution1.firstMissingPositive(nums3));
    }
}

File: cpp/unittests/FirstMissingPositiveTest.cpp
#include "FirstMissingPositive.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(FirstMissingPositive, Solution41_1) {
  std::vector<int> Nums1{1, 2, 0};
  const int Expected1 = 3;
  EXPECT_EQ(Expected1, Solution41_1().firstMissingPositive(Nums1));

  std::vector<int> Nums2{3, 4, -1, 1};
  const int Expected2 = 2;
  EXPECT_EQ(Expected2, Solution41_1().firstMissingPositive(Nums2));

  std::vector<int> Nums3{7, 8, 9, 11, 12};
  const int Expected3 = 1;
  EXPECT_EQ(Expected3, Solution41_1().firstMissingPositive(Nums3));
}